CHANGELOG.md

```diff
@@ -88,6 +88,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - [Workload Management] Make query groups persistent across process restarts [#16370](https://github.com/opensearch-project/OpenSearch/pull/16370)
 - Fix inefficient Stream API call chains ending with count() ([#15386](https://github.com/opensearch-project/OpenSearch/pull/15386))
 - Fix array hashCode calculation in ResyncReplicationRequest ([#16378](https://github.com/opensearch-project/OpenSearch/pull/16378))
+- Compare strings with equals() instead of == ([#16415](https://github.com/opensearch-project/OpenSearch/pull/16415))
```

Comment:
Let's be less specific in the change log.  "Fix string equality comparisons" is probably good enough.